app/ante/ante.go

```diff
@@ -177,9 +182,35 @@ func IsSystemTx(tx sdk.Tx, isAuthorizedSigner func(string) bool) bool {
 		*observertypes.MsgAddBlameVote:
 		signers := innerMsg.GetSigners()
 		if len(signers) == 1 {
-			return isAuthorizedSigner(signers[0].String())
+			if isObserverAuthorized(signers[0].String()) {
+				return true
+			}
 		}
 	}
 
 	return false
 }
+
+// IsAuthorityTx checks if there is a MsgCreateValidator in the transaction and returns the signer if present.
+func IsAuthorityTx(tx sdk.Tx) (bool, string) {
+	for _, msg := range tx.GetMsgs() {
+		switch msg.(type) {
+		case *stakingtypes.MsgCreateValidator: // Other message types can be added here
+			signers := msg.GetSigners()
```

Comment:
as I mentioned before, just 1 case is does not make sense to using switch statement.

```suggestion
		if validatorMsg, ok := msg.(*stakingtypes.MsgCreateValidator); ok {
			signers := validatorMsg.GetSigners()```

Reply:
I adopted the switch statement in anticipation of additional Msg types to be managed as AuthorityTx in the future.

Reply:
Yes, I know what the intention is ;)
But it is easier to read simple code with a minimum of requirements.
It can also be added when such a situation arises.